### PR TITLE
ci: move selenium, playwright and license check tests to GHA 

### PR DESCRIPTION
**Pull request by Zidious**
This PR moves Selenium, Playwright and license check to GHA.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            - ./selenium/node_modules
+            - ./playwright/node_modules
+          key: npm-workspace-cache-${{ runner.os }}-${{ hashFiles('./**/package-lock.json') }}
+          restore-keys: |
+            npm-workspace-cache-${{ runner.os }}-
+      - name: Cache Maven dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
```
Comment (stephenmathieson) [inline .github/workflows/tests.yml:29]: Does the Java version matter? Do dependencies for Java v8 work with Java v20?

```suggestion
          key: ${{ runner.os }}-maven-${{ matrix.java }}-${{ hashFiles('**/pom.xml') }}
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Start fixture server
+        run: npm --prefix=playwright start &
+      - name: Run Playwright tests
+        run: mvn test -pl playwright
+      - name: Run Selenium tests
+        run: selenium/node_modules/axe-test-fixtures/fixtures && python2 -m SimpleHTTPServer 8001 & mvn test -q -pl selenium
```
Comment (stephenmathieson) [inline .github/workflows/tests.yml:58]: Should this be:

```suggestion
        run: python2 -m SimpleHTTPServer 8001 & mvn test -q -pl selenium
        working-directory: selenium/node_modules/axe-test-fixtures/fixtures
```

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  selenium-tests:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Start fixture server
+        run: selenium/node_modules/axe-test-fixtures/fixtures &
```
Comment (stephenmathieson) [inline .github/workflows/tests.yml:66]: Is `fixtures` a program?

Comment (stephenmathieson) [reply .github/workflows/tests.yml:69]: [It doesn't look like it](https://github.com/dequelabs/axe-test-fixtures). I think this should be:

```suggestion
        run:  python2 -m SimpleHTTPServer 8001 &
        working-directory: selenium/node_modules/axe-test-fixtures/fixtures
```

File: .github/workflows/tests.yml (hunk 1 above)
Comment (Zidious) [reply .github/workflows/tests.yml:31]: Ya this is a draft (I accidentally opened it up when turned it into a draft so you'll still get spammed in GH notifications :( ) - We can just use Java 8 no need to waste build minutes here

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Cache npm dependencies
+        uses: actions/cache@v3
+        id: npm-cache
+        with:
+          path: |
+            node_modules
+            selenium/node_modules
+            playwright/node_modules
```
Comment (stephenmathieson) [inline .github/workflows/tests.yml:22]: Can just do:

```suggestion
          path: '**/node_modules'
```

File: selenium/pom.xml [hunk 5]
```diff
@@ -175,7 +175,7 @@
     <dependency>
       <groupId>org.seleniumhq.selenium</groupId>
       <artifactId>selenium-java</artifactId>
-      <version>4.7.2</version>
+      <version>4.10.0</version>
```
Comment (Zidious) [inline selenium/pom.xml:178]: Using localhost caused 403 on > chromedriver 111 https://github.com/SeleniumHQ/selenium/issues/11750#issuecomment-1460580132

Updating to 4.10.0 resolved this - Can move out of this PR if required

File: .circleci/license-check.js [hunk 6]
```diff
@@ -1,11 +1,11 @@
-const fs = require('fs');
```
Comment (stephenmathieson) [inline .circleci/license-check.js:1]: IMO this should no longer be in the `.circleci/` directory.

**Comment by michael-siek** [comment]
I think we should also move releases to GHA. Right now releases are blocked until test pass.

**Comment by Zidious** [comment]
> I think we should also move releases to GHA. Right now releases are blocked until test pass.

They shouldn't be, I've removed tests and license checks from the CircleCI release job and now we exclusively run tests on branches that are no develop or master (no need to waste minutes running tests there)

**Review by michael-siek — APPROVED** [review]

**Pull request merged**